src/logger/src/logger.rs

```diff
@@ -432,345 +173,186 @@ impl Log for Logger {
+        assert_eq!(LevelFilter::from_str("Debug"), Ok(LevelFilter::Debug));
+        assert_eq!(LevelFilter::from_str("Info"), Ok(LevelFilter::Info));
+        assert_eq!(LevelFilter::from_str("Warn"), Ok(LevelFilter::Warn));
+        assert_eq!(LevelFilter::from_str("Error"), Ok(LevelFilter::Error));
 
-        // Check show_level=false.
-        logger
-            .set_include_level(false)
-            .set_include_origin(true, true);
-        logger.mock_log(Level::Info, "msg");
-        validate_log(
-            &mut Box::new(&mut reader),
-            &format!("[{}:logger.rs:0] msg\n", crnt_thread_name),
-        );
+        assert_eq!(LevelFilter::from_str("OFF"), Ok(LevelFilter::Off));
+        assert_eq!(LevelFilter::from_str("TRACE"), Ok(LevelFilter::Trace));
+        assert_eq!(LevelFilter::from_str("DEBUG"), Ok(LevelFilter::Debug));
+        assert_eq!(LevelFilter::from_str("INFO"), Ok(LevelFilter::Info));
+        assert_eq!(LevelFilter::from_str("WARN"), Ok(LevelFilter::Warn));
+        assert_eq!(LevelFilter::from_str("ERROR"), Ok(LevelFilter::Error));
 
-        // Test with a mock instance id.
-        logger.set_instance_id(TEST_INSTANCE_ID.to_string());
-
-        // Check that the prefix contains only the instance id when all flags are false.
-        logger
-            .set_include_level(false)
-            .set_include_origin(false, false);
-        logger.mock_log(Level::Info, "msg");
-        validate_log(
-            &mut Box::new(&mut reader),
-            &format!("[TEST-INSTANCE-ID:{}] msg\n", crnt_thread_name),
-        );
-
-        // Check that the prefix is correctly shown when all flags are true.
-        logger
-            .set_include_level(true)
-            .set_include_origin(true, true);
-        logger.mock_log(Level::Warn, "msg");
-        validate_log(
-            &mut Box::new(&mut reader),
-            &format!(
-                "[TEST-INSTANCE-ID:{}:WARN:logger.rs:0] msg\n",
-                crnt_thread_name
-            ),
-        );
+        assert_eq!(LevelFilter::from_str("Warning"), Ok(LevelFilter::Warn));
+        assert_eq!(LevelFilter::from_str("WARNING"), Ok(LevelFilter::Warn));
     }
 
     #[test]
-    fn test_thread_name_custom() {
-        let custom_thread = thread::Builder::new()
-            .name("custom-thread".to_string())
-            .spawn(move || {
-                let logger = Logger::mock_new();
-                let mut reader = logger.mock_init();
-                logger
-                    .set_include_level(false)
-                    .set_include_origin(false, false);
-                logger.set_instance_id("".to_string());
-                logger.mock_log(Level::Info, "thread-msg");
-                validate_log(&mut Box::new(&mut reader), "[custom-thread] thread-msg\n");
-            })
+    fn logger() {
+        // Create temp file.
+        let id = uuid::Uuid::new_v4();
+        let path = format!("/tmp/{id}");
```

Comment:
Here you can use a temporary file instead of a UUID. See https://docs.rs/vmm-sys-util/latest/vmm_sys_util/tempfile/index.html

It should be accessible from `utils::tempfile::TempFile`

Reply:
I don't follow... since it's in scope, it should only delete itself at the end of the test no?

Reply:
To pass the `temp_file` to the logger it needs to be a `File` [`TempFile::into_file`](https://docs.rs/vmm-sys-util/latest/vmm_sys_util/tempfile/struct.TempFile.html#method.into_file) removes the file from the file system making it unusable.

So to expand on my example it would need to be:
```rust
let temp_file = TempFile::new();
let path = temp_file.as_path();
std::mem::forget(temp_file); // This also leaks a small amount of memory for the rest of this test run.
let file = std::fs::OpenOptions::new().write(true).open(&path).unwrap();
// ...
std::fs::remove(path).unmwrap();
```

Reply:
I haven't been able to get a version working quickly using `TempFIle` something that you might expect to work like:
```rust
        // Create temp file.
        let target = TempFile::new().unwrap();
        let mut file = target.into_file();

        // Create logger.
        let logger = Logger(Mutex::new(LoggerConfiguration {
            target: Some(file.try_clone().unwrap()),
            format: LogFormat {
                show_level: true,
                show_log_origin: true,
            },
        }));

        // Assert results of enabled given specific metadata.
        assert!(logger.enabled(&Metadata::builder().level(Level::Warn).build()));
        assert!(logger.enabled(&Metadata::builder().level(Level::Debug).build()));

        // Log
        let metadata = Metadata::builder().level(Level::Error).build();
        let record = Record::builder()
            .args(format_args!("Error!"))
            .metadata(metadata)
            .file(Some("dir/app.rs"))
            .line(Some(200))
            .module_path(Some("module::server"))
            .build();
        logger.log(&record);

        Assert calling flush panics.
        std::panic::catch_unwind(|| {
            logger.flush();
        })
        .unwrap_err();

        // Asserts result of log.
        let mut contents = String::new();
        file.read_to_string(&mut contents).unwrap();
        let (_time, rest) = contents.split_once(' ').unwrap();
        let thread = thread::current().name().unwrap_or("-").to_string();
        assert_eq!(
            rest,
            format!("[{DEFAULT_INSTANCE_ID}:{thread}:ERROR:dir/app.rs:200] Error!\n")
        );
```
doesn't.

Reply:
Yeah that could be improved, but there are existing alternatives, so let's use those rather than introducing a new pattern to creating temp files and a new dependency.

Reply:
Okay.

I think there is also a race condition where inbetween `temp_file.remove()` and `OpenOptions::new().write(true).open(path).unwrap()` the system could create the file with `path` for another `TempFile`. But it is very unlikely we encounter this in unittests.

Reply:
Why can't we do `let tmp_file = TempFile::new(); target: Some(tmp_file.as_file().clone())`?

Reply:
> Why can't we do `let tmp_file = TempFile::new(); target: Some(tmp_file.as_file().clone())`? `TempFile::as_file` doesnt drop the `TempFile` and thus doesnt delete it, so it should work.

This is the approach I tried above with `as_file().try_clone()` but I couldn't get it working.